src/GitVersionCore/GitVersionCacheKeyFactory.cs

```diff
@@ -23,10 +24,112 @@ private static string GetGitSystemHash(GitPreparer gitPreparer, IFileSystem file
+            var result = new List<string>();
+            result.Add(root);
+            // Data structure to hold names of subfolders to be
+            // examined for files.
+            Stack<string> dirs = new Stack<string>(20);
+
+            if (!System.IO.Directory.Exists(root))
+            {
+                throw new ArgumentException();
+            }
+
+            dirs.Push(root);
+
+            while (dirs.Count > 0)
+            {
+                string currentDir = dirs.Pop();
+
+                var di = new DirectoryInfo(currentDir);
+                result.Add(di.Name);
+
+                string[] subDirs;
+                try
+                {
+                    subDirs = System.IO.Directory.GetDirectories(currentDir);
+                }
+                // An UnauthorizedAccessException exception will be thrown if we do not have
+                // discovery permission on a folder or file. It may or may not be acceptable 
+                // to ignore the exception and continue enumerating the remaining files and 
+                // folders. It is also possible (but unlikely) that a DirectoryNotFound exception 
+                // will be raised. This will happen if currentDir has been deleted by
+                // another application or thread after our call to Directory.Exists. The 
+                // choice of which exceptions to catch depends entirely on the specific task 
+                // you are intending to perform and also on how much you know with certainty 
+                // about the systems on which this code will run.
+                catch (UnauthorizedAccessException e)
+                {
+                    Logger.WriteError(e.Message);
+                    continue;
+                }
+                catch (System.IO.DirectoryNotFoundException e)
+                {
+                    Logger.WriteError(e.Message);
+                    continue;
+                }
+
+                string[] files = null;
+                try
+                {
+                    files = System.IO.Directory.GetFiles(currentDir);
+                }
+
+                catch (UnauthorizedAccessException e)
+                {
+                    Logger.WriteError(e.Message);
+                    continue;
+                }
+
+                catch (System.IO.DirectoryNotFoundException e)
+                {
+                    Logger.WriteError(e.Message);
+                    continue;
+                }
+                // Perform the required action on each file here.
+                // Modify this block to perform your required task.
+                foreach (string file in files)
+                {
+                    try
+                    {
+                        // Perform whatever action is required in your scenario.
+                        System.IO.FileInfo fi = new System.IO.FileInfo(file);
+                        result.Add(fi.Name);
+
+                        var reader = fi.OpenText();
```

Comment:
It does traverse. 

> On Jul 22, 2016, at 02:11, Daniel Rose <notifications@github.com> wrote:
> 
> In src/GitVersionCore/GitVersionCacheKeyFactory.cs:
> 
> > +                catch (System.IO.DirectoryNotFoundException e)
> > +                {
> > +                    Logger.WriteError(e.Message);
> > +                    continue;
> > +                }
> > +                // Perform the required action on each file here.
> > +                // Modify this block to perform your required task.
> > +                foreach (string file in files)
> > +                {
> > +                    try
> > +                    {
> > +                        // Perform whatever action is required in your scenario.
> > +                        System.IO.FileInfo fi = new System.IO.FileInfo(file);
> > +                        result.Add(fi.Name);
> > +
> > +                        var reader = fi.OpenText();
> You can simplify this by using results.Add(File.ReadAllText(file))
> 
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub, or mute the thread.
>